From 79846b358f6ea8d0a6027ee0ba38342a730065b0 Mon Sep 17 00:00:00 2001
From: Roy Marples <roy@marples.name>
Date: Sun, 28 Jul 2019 11:49:17 +0100
Subject: script: ensure that tmp files are removed

fopen creates a new stream which will use a new fd.
Why using fdopen, the new stream will use the same fd, so when
it's closed the tmp file will really unlink.
---
 src/script.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/script.c b/src/script.c
index 1863a661..74aef1b1 100644
--- a/src/script.c
+++ b/src/script.c
@@ -213,10 +213,11 @@ make_env(const struct interface *ifp, const char *reason)
 	if (tmpfd == -1)
 		goto eexit;
 	unlink(tmpfile);
-	fp = fopen(tmpfile, "w+");
-	close(tmpfd);
-	if (fp == NULL)
+	fp = fdopen(tmpfd, "w+");
+	if (fp == NULL) {
+		close(tmpfd);
 		goto eexit;
+	}
 #endif
 
 #ifdef INET
-- 
cgit v1.2.1

